api/src/model/request/play_asset_pack_fetch_request.gd

```diff
@@ -0,0 +1,130 @@
+# ##############################################################################
+#
+#	Copyright 2020 Google LLC
+#
+#	Licensed under the Apache License, Version 2.0 (the "License");
+#	you may not use this file except in compliance with the License.
+#	You may obtain a copy of the License at
+#
+#		https://www.apache.org/licenses/LICENSE-2.0
+#
+#	Unless required by applicable law or agreed to in writing, software
+#	distributed under the License is distributed on an "AS IS" BASIS,
+#	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#	See the License for the specific language governing permissions and
+#	limitations under the License.
+#
+# ##############################################################################
+#
+# Request object that handles asynchronous logic related to fetch_asset_pack().
+#
+# This object provides relevant getters so that it is possible to retrieve
+# the updated states from this object upon every state update.
+#
+# ##############################################################################
+class_name PlayAssetPackFetchRequest
+extends PlayAssetDeliveryRequest
+
+# -----------------------------------------------------------------------------
+# Emits request_completed(did_succeed, pack_name, result, exception) signal 
+# upon request succeeds(reaching CANCELED/COMPLETED)/fails(reaching FAILED).
+# 	did_succeed : boolean indicating request succeeded/failed
+# 	pack_name: String, name of the requested asset pack
+# 	result : PlayAssetPackState object
+#	exception: PlayAssetPackException object if Plugin encountered an exception
+# while handling this request
+# -----------------------------------------------------------------------------
+signal request_completed(did_succeed, pack_name, result, exception)
```

Comment:
I think ```did_succeed``` arg can be remove in this case. ```Exception``` field is kind of different since it is a PlayAssetPackException object, and can not be easily retrieved without reference to the PlayAssetDeliveryRequest object.